testbeam_analysis/result_analysis.py

```diff
@@ -848,3 +849,56 @@ def calculate_efficiency(input_tracks_file, input_alignment_file, output_pdf, bi
                         out_pass[:] = total_track_density_with_DUT_hit.T
                         out_total[:] = total_track_density.T
     return efficiencies, pass_tracks, total_tracks
+
+
+def track_angle(input_tracks_file, output_track_angle_file, use_duts=None, output_pdf=None):
+    '''Calculates and histograms the track angle of the fitted tracks for selected DUTs.
+    Parameters
+    ----------
+    input_tracks_file : string
+        file name with the tracks table
+    output_track_angle_file: string
+        output pytables file with fitted means and sigmas of track angles for selected DUTs
+    use_duts : iterable
+        The duts to plot/calculate the track angle. If None all duts in the input_tracks_file are used
+    output_pdf : string
+        file name for the output pdf
+    '''
+    logging.info('=== Calculate track angles ===')
+
+    slopes = {}
+    with tb.open_file(input_tracks_file, 'r') as in_file_h5:
+        n_duts = len(in_file_h5.list_nodes("/"))  # determine number of DUTs
+
+        if use_duts is None:
+            use_duts = range(n_duts)
+        else:
+            use_duts = use_duts
+        dut_index = [('Tracks_DUT_%i' % i) for i in use_duts]
+
+        widgets = ['', progressbar.Percentage(), ' ',
+                   progressbar.Bar(marker='*', left='|', right='|'),
+                   ' ', progressbar.AdaptiveETA()]
+        progress_bar = progressbar.ProgressBar(widgets=widgets,
+                                               maxval=len(use_duts),
+                                               term_width=80)
+        progress_bar.start()
+
+        i = 0  # counting number of progressed DUTs, needed for progressbar
+        for node in in_file_h5.root:  # loop through all DUTs in track table
+            if node.name in dut_index:  # only store track slopes of selected DUTs
+                slopes.update({node.name: [node[::10]['slope_0'],
```

Comment:
This is a little hackish. You just load every 10th track into RAM. No guarantee that it fits and a lot of tracks a not used. Imagine you only have 100 tracks. Please take the complete table. This table should be read in chunks as it is done in the [other](https://github.com/SiLab-Bonn/testbeam_analysis/blob/development/testbeam_analysis/result_analysis.py#L709) result analysis function. There is a helper function `data_aligned_at_events` that you could use. Although event aligning is not really needed here.